api_manager_health.go

```diff
@@ -0,0 +1,92 @@
+/* Copyright © 2020 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: BSD-2-Clause
+
+   Generated by: https://github.com/swagger-api/swagger-codegen.git */
+
+package nsxt
+
+import (
+	"context"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/vmware/go-vmware-nsxt/administration"
+)
+
+// Linger please
+var (
+	_ context.Context
+)
+
+type NsxManagerHealthApiService service
+
+/*
+NsxManagerHealthApiService Gets Node health status
+Returns 200 if node is healthy and 503 if node is not
+ * @param ctx context.Context - for authentication, logging, cancellation, deadlines, tracing, etc. Passed from http.Request or context.Background().
+
+@return NodeHealth
+*/
+func (a *NsxManagerHealthApiService) GetNodeHealth(ctx context.Context) (administration.NodeHealth, *http.Response, error) {
+	var (
+		localVarHttpMethod  = strings.ToUpper("Get")
+		localVarPostBody    interface{}
+		localVarFileName    string
+		localVarFileBytes   []byte
+		localVarReturnValue administration.NodeHealth
+	)
+
+	// create path and map variables
+	localVarPath := a.client.cfg.BasePath + "/reverse-proxy/node/health"
+
+	localVarHeaderParams := make(map[string]string)
+	localVarQueryParams := url.Values{}
+	localVarFormParams := url.Values{}
+
+	// to determine the Content-Type header
+	localVarHttpContentTypes := []string{"application/json"}
+
+	// set Content-Type header
+	localVarHttpContentType := selectHeaderContentType(localVarHttpContentTypes)
+	if localVarHttpContentType != "" {
+		localVarHeaderParams["Content-Type"] = localVarHttpContentType
+	}
+
+	// to determine the Accept header
+	localVarHttpHeaderAccepts := []string{"application/json"}
+
+	// set Accept header
+	localVarHttpHeaderAccept := selectHeaderAccept(localVarHttpHeaderAccepts)
+	if localVarHttpHeaderAccept != "" {
+		localVarHeaderParams["Accept"] = localVarHttpHeaderAccept
+	}
+	r, err := a.client.prepareRequest(ctx, localVarPath, localVarHttpMethod, localVarPostBody, localVarHeaderParams, localVarQueryParams, localVarFormParams, localVarFileName, localVarFileBytes)
+	if err != nil {
+		return localVarReturnValue, nil, err
+	}
+
+	localVarHttpResponse, err := a.client.callAPI(r)
+	if err != nil || localVarHttpResponse == nil {
+		return localVarReturnValue, localVarHttpResponse, err
+	}
+
+	localVarBody, err := ioutil.ReadAll(localVarHttpResponse.Body)
+	localVarHttpResponse.Body.Close()
+	if err != nil {
+		return localVarReturnValue, localVarHttpResponse, err
+	}
+
+	if localVarHttpResponse.StatusCode == 200 || localVarHttpResponse.StatusCode == 503 {
```

Comment:
do we expect `administration.NodeHealth` format in case on 503?

Reply:
Thanks for the comment! Yes, it will return NodeHeath which carries field "healthy" with value false in case of 503.